source/server/listener_manager_impl.cc

```diff
@@ -223,21 +223,37 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
     addListenSocketOptions(
         Network::SocketOptionFactory::buildLiteralOptions(config.socket_options()));
   }
+  bool is_quic{false};
+
   if (socket_type_ == Network::Address::SocketType::Datagram) {
     // Needed for recvmsg to return destination address in IP header.
     addListenSocketOptions(Network::SocketOptionFactory::buildIpPacketInfoOptions());
     // Needed to return receive buffer overflown indicator.
     addListenSocketOptions(Network::SocketOptionFactory::buildRxQueueOverFlowOptions());
-    std::string listener_name =
-        config.has_udp_listener_config() ? config.udp_listener_config().udp_listener_name() : "";
-    if (listener_name.empty()) {
-      listener_name = UdpListenerNames::get().RawUdp;
+    const auto& udp_config = config.has_udp_listener_config()
+                                 ? config.udp_listener_config()
+                                 : envoy::api::v2::listener::UdpListenerConfig();
+    const std::string listener_name = udp_config.udp_listener_name().empty()
+                                          ? UdpListenerNames::get().RawUdp
+                                          : udp_config.udp_listener_name();
+    is_quic = listener_name == UdpListenerNames::get().Quic;
+    try {
+      udp_listener_factory_ =
+          Config::Utility::getAndCheckFactory<ActiveUdpListenerConfigFactory>(listener_name)
+              .createActiveUdpListenerFactory(config.has_udp_listener_config()
+                                                  ? config.udp_listener_config()
+                                                  : envoy::api::v2::listener::UdpListenerConfig());
+    } catch (EnvoyException ex) {
+      // TODO(danzh): add implementation for quic_listener and do not catch
```

Comment:
I'm confused. Why do we need/want this fallback?

Reply:
Can we just assert this then or wait until the other one is merged?

Reply:
I'll wait for the other PR merge first and change this line. It is as such now to make listener mgr test pass in a meaningful way.